### PR TITLE
CA-523-Please-remove-the-reporter's-remove-parties-function-

### PR DESCRIPTION
**Pull request by giladadon1997**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [AHOYAPPS-0000](https://issues.corp.twilio.com/browse/AHOYAPPS-0000)

### Description

A description of what this PR does.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

File: src/components/MenuBar/MenuBar.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import FormControl from '@material-ui/core/FormControl';
 import InputLabel from '@material-ui/core/InputLabel';
 import { Offline, Online } from 'react-detect-offline';
 import { TRACK_TYPE, NOTIFICATION_MESSAGE, ERROR_MESSAGE } from '../../utils/displayStrings';
-import { PARTICIANT_TYPES } from '../../utils/participantTypes';
+import { DOE_PARTICIPANT_TYPES } from '../../utils/rbac/DOEParticipantTypes';
```
Comment (talr) [inline src/components/MenuBar/MenuBar.tsx]: I prefer not to have DOE or any customer in our code, let's make it generic product / service.
DOE should be only in the database.

File: src/components/MenuBar/MenuBar.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ const useStyles = makeStyles(theme =>
   })
 );
 const getPartyTypes = () => {
-  return Object.values(PARTICIANT_TYPES);
+  return Object.values(DOE_PARTICIPANT_TYPES);
```
Comment (talr) [inline src/components/MenuBar/MenuBar.tsx]: Same

File: src/hooks/useDataTrackListener/useDataTrackListener.tsx [hunk 3]
```diff
@@ -3,15 +3,21 @@ import { useEffect } from 'react';
...
 import { IMuteRemoteParticipantMessage } from '../../utils/muteRemoteParticipantMessage';
 
 export default function useDataTrackListener() {
   const { room, localTracks } = useVideoContext();
 
   useEffect(() => {
     const handleRemoteParticipant = (participant: RemoteParticipant) => {
-      if (isModerator(ParticipantIdentity.Parse(participant.identity).partyType)) {
+      if (
+        roleChecker.doesRoleHavePermission(
```
Comment (talr) [inline src/hooks/useDataTrackListener/useDataTrackListener.tsx:16]: Based on the Spreadsheet I shared with you I think there should be three parameters or this function is only about the permissions of the person who is running it.
I think we should also have function with three parameters to query who can do what?

File: src/hooks/useIsHostIn/useIsHostIn.tsx [hunk 4]
```diff
@@ -45,13 +45,13 @@ export default function useIsHostIn() {
...
       theRoom.participants.forEach(participant => {
-        if (ParticipantIdentity.Parse(participant.identity).partyType === PARTICIANT_TYPES.REPORTER) {
+        if (ParticipantIdentity.Parse(participant.identity).partyType === DOE_PARTICIPANT_TYPES.REPORTER) {
           flag = true;
         }
       });
       if (
-        ParticipantIdentity.Parse(theRoom.localParticipant.identity).partyType === PARTICIANT_TYPES.REPORTER ||
-        ParticipantIdentity.Parse(theRoom.localParticipant.identity).partyType === PARTICIANT_TYPES.HEARING_OFFICER
+        ParticipantIdentity.Parse(theRoom.localParticipant.identity).partyType === DOE_PARTICIPANT_TYPES.REPORTER ||
```
Comment (talr) [inline src/hooks/useIsHostIn/useIsHostIn.tsx]: DOE Reporter should come from the database
Few options:
1. Info(UserId=124, Customer=DOE) Return a structure (i.e. security clearance, skillset ...)
2. Is( Customer=DOE, Roles=Reporter) returns Boolean

File: src/hooks/useIsHostIn/useIsHostIn.tsx [hunk 5]
```diff
@@ -66,11 +66,11 @@ export default function useIsHostIn() {
     if (theRoom !== null && typeof theRoom.participants !== 'undefined') {
```
Comment (talr) [inline src/hooks/useIsHostIn/useIsHostIn.tsx:66]: Let's use the same constant in every place - so we can change the constant value and everything will continue to work

**Review by talr — CHANGES_REQUESTED** [review]
Looks good! 
Please address my change requests.
Thanks,
Tal

File: src/components/MenuBar/MenuBar.tsx (hunk 1 above)
Comment (gila1997) [reply src/components/MenuBar/MenuBar.tsx]: I replaced it to PARTICIANT_TYPES

File: src/hooks/useDataTrackListener/useDataTrackListener.tsx (hunk 3 above)
Comment (gila1997) [reply src/hooks/useDataTrackListener/useDataTrackListener.tsx:16]: Not all cases have three parameters, here there are only 2

File: src/hooks/useIsHostIn/useIsHostIn.tsx (hunk 5 above)
Comment (gila1997) [reply src/hooks/useIsHostIn/useIsHostIn.tsx:66]: I can make it a constant.
But that's not how you write in  React,

File: src/components/MenuBar/MenuBar.tsx (hunk 1 above)
Comment (talr) [reply src/components/MenuBar/MenuBar.tsx]: I still see DOE in the code, can you please remove it and make it generic.

**Review by talr — CHANGES_REQUESTED** [review]
I don't see the changes - I still see DOE in the PR code.
I don't see feedback to all the comments in the Conversation.
Have you implemented all the Privileges Spreadsheet I shared with in the RBAC function?
Can we move the RBAC methods to a separate repository?

File: src/hooks/useIsHostIn/useIsHostIn.tsx (hunk 4 above)
Comment (giladadon1997) [reply src/hooks/useIsHostIn/useIsHostIn.tsx]: let's talk in code review, I didn't understand what you want to say

File: src/components/ParticipantInfo/ParticipantDropDown/ParticipantDropDown.tsx [hunk 6]
```diff
@@ -5,17 +5,41 @@ import MenuItem from '@material-ui/core/MenuItem';
...
+    let remoteParticipantPartyType = ParticipantIdentity.Parse(participant.identity).partyType;
+    if (localParticipantType === remoteParticipantPartyType) return options;
+    if (
+      roleChecker.doesRoleHavePermission(
+        ROLE_PERMISSIONS.MUTE_PARTICIPANT,
+        localParticipantType,
+        remoteParticipantPartyType
+      )
+    )
+      options[0] = MUTE;
```
Comment (talr) [inline src/components/ParticipantInfo/ParticipantDropDown/ParticipantDropDown.tsx]: Use push()

**Review by talr — APPROVED** [review]

**Pull request merged**